BZ-18845: feat: select component

anjalideswal opened the pull request:
- added showDropDownArrow optional property to toggle arrow visbility
- added dynamic css variables for box-shadow, hover background-color, selected-option-icon

Dev proof - https://drive.google.com/file/d/1QSu0lkBobQh6YFMqDltyLcWt_vN_lZIG/view?usp=sharing

murdore commented on src/lib/Select/Select.svelte:
```diff
@@ -155,11 +155,13 @@
-      />
+      {#if properties.showDropDownArrow}
+        <img
+          src="https://sdk.breeze.in/gallery/icons/right-arrow.svg"
```
can we pass this as input? also expose the alt tag to be passed by component user

anjalideswal replied:
Done. Added `dropDownIcon` as optional prop. If not passed, drop down icon will not be added, to handle drop down icon visibility. Added altText export

sinha-sahil commented:
@anjalideswal This seems to be more of a enhancement than a fix. Kindly, update the type.

rab1prasad commented on src/lib/Select/Select.svelte:
```diff
@@ -5,6 +5,7 @@
   import Button from '$lib/Button/Button.svelte';
 
   let selectedElementDiv: HTMLDivElement | null = null;
+  export let altText = '';
```
rename, please specify for whom this alt tex is for.

rab1prasad commented on src/lib/Select/properties.ts:
```diff
@@ -6,4 +6,5 @@ export type SelectProperties = {
   selectedItemLabel: string | string[] | null;
   showSelectedItemInDropdown: boolean;
   selectMultipleItems: boolean;
+  dropDownIcon?: string | null;
```
Do we need this `?` if using null?

I think we should have 2 different props one for `showDropDownIcon` one for dropdown icon, and a default icon should be provided if icon is not passed as prop.

murdore commented on src/lib/Select/Select.svelte:
```diff
@@ -155,11 +158,13 @@
-        class="arrow {isSelectOpen ? 'active' : ''}"
-      />
+      {#if !properties.hideDropDownIcon}
+        <img
```
@anjalideswal lets use the Image component from svelteComponents.
Check @adarshjuspay 's pr for reference

murdore commented:
@anjalideswal resolve conflicts and add proof of testing

Commit:
BZ-18845: feat: select component

- added showDropDownArrow optional property to toggle arrow visbility
- added dynamic css variables for box-shadow, hover background-color, selected-option-icon
